training/SYMBOL_TABLE_OVERVIEW.md

```diff
@@ -0,0 +1,150 @@
+
+```chapel
+  var entryList = st.findAll("dataset_entry_/d+");
+  // entryList: [`dataset_entry_0`, `dataset_entry_1`, 'dataset_entry_2']
+```
+
+You can then loop through the returned array of names to access the data and perform operations on each entry in the set.
+
+<a id="accessSym"></a>
+## Accessing Symbol Table Entry Data
+
+Depending on your functionality requirements, there are a couple different ways to access a symbol table entry.
+
+### Passing the Entry to the Client
+
+If you need to return an entry to the python for computations, the `SymTab.attrib(name)` method will return the entry's attributes, which are required by the arkouda client to reference the entry and use the underlying data. You can then use the return message to `attach` to the entry and access the data as you would with any other Python object.
+
+More information on passing data between the client and server can be found [here](https://github.com/Bears-R-Us/arkouda/blob/master/training/MESSAGING_OVERVIEW.md).
+
+### Working with Entries in Chapel
+
+When working with a basic `SymEntry` entry type, you can easily access the data values within the entry by accessing the property `a`. To access the domain information, access the property `aD`. For example, assume we have a SymTab entry variable called `entry`.
+
+```chapel
+  var values = entry.a;
+  var domain = entry.aD;
+```
+
+Each entry type has its own set of properties that can be accessed using the `.` notation, similar to accessing class properties in Python objects. Below is a list of all accessible properties for each entry type within the hierarchy:
+
+- `AbstractSymEntry`
+  - `entryType`
+  - `assignableTypes`
+  - `name`
+  - `GenSymEntry`
+    - All properties of `AbstractSymEntry`
+    - `dtype`
+    - `itemsize`
+    - `size`
+    - `ndim`
+    - `shape`
+    - `SymEntry`
+      - All properties of `GenSymEntry`
+      - `etype` - Chapel's version of `dtype`
+      - `a` - Stores the values of the array
+      - `aD` - Stores the domain information
+    - `SegArraySymEntry`
+      - All properties of `GenSymEntry`
+      - `segmentsEntry`
+      - `valuesEntry`
+      - `lengthsEntry`
+    - `SegStringSymEntry`
+      - All properties of `GenSymEntry`
+      - `offsetsEntry`
+      - `bytesEntry`
+  - `CompositeSymEntry`
+    - All properties of `AbstractSymEntry`
+    - `ndim`
+    - `size`
+    - `GroupBySymEntry`
+      - All properties of `CompositeSymEntry`
+      - `keyNamesEntry`
+      - `keyTypesEntry`
+      - `segmentsEntry`
+      - `permEntry`
+      - `ukIndEntry`

```

Comment:
I would break this out instead of also having the entries tiered:

- AbstractSymEntry
    - Properties listed
- GenSymEntry
    - Properties listed
etc.

I do like how you reference the properties of the parent.

Reply:
I agree, the current way makes it hard to tell where the properties end and the child class begins once rendered
- `AbstractSymEntry`
  - `entryType`
  - `assignableTypes`
  - `name`
  - `GenSymEntry`
    - All properties of `AbstractSymEntry`
    - `dtype`
    - `itemsize`
    - `size`
    - `ndim`
    - `shape`
    - `SymEntry`
      - All properties of `GenSymEntry`

Reply:
Reformatted to be on one level with spacing. I think it's clearer now, but you can be the judge